api/src/main/java/org/pmiops/workbench/cohorts/CohortMaterializationService.java

```diff
@@ -47,11 +45,48 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.Nullable;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 @Service
 public class CohortMaterializationService {
 
-  private static final String DOMAIN_CONCEPT_STANDARD = "standard";
-  private static final String DOMAIN_CONCEPT_SOURCE = "source";
+  // See https://cloud.google.com/bigquery/docs/parameterized-queries for JSON configuration of query parameters
+  private static final Function<Map.Entry<String, QueryParameterValue>, Map<String, Object>> TO_QUERY_PARAMETER_MAP =
+      (entry) -> {
+        Map<String, Object> result = new HashMap<>();
```

Comment:
opt_nit: I think it would be slightly more readable to put this at the end like:

```
return ImmutableMap.newBuilder()
  .put("name", entry.getKey())
  .put(...)
  .build();
```

Reply:
Switched to use builder everywhere in here. Not sure if this is actually more readable, but at least it's immutable. :)